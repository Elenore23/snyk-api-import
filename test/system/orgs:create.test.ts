import { exec } from 'child_process';
import * as path from 'path';
const main = './dist/index.js'.replace(/\//g, path.sep);

describe('`snyk-api-import help <...>`', () => {
  const OLD_ENV = process.env;

  afterAll(async () => {
    process.env = { ...OLD_ENV };
  });
  it('Shows help text as expected', async (done) => {
    return exec(`node ${main} orgs:create help`, (err, stdout) => {
      if (err) {
        throw err;
      }
      expect(err).toBeNull();
      expect(stdout.trim()).toMatchSnapshot();
      done();
    });
  });
  it('Fails to create an org as expected', async (done) => {
    const pathToBadJson = path.resolve(
      __dirname + '/fixtures/create-orgs/fails-to-create/1-org.json',
    );
    const logPath = path.resolve(
      __dirname + '/fixtures/create-orgs/fails-to-create/',
    );

    return exec(
      `node ${main} orgs:create --file=${pathToBadJson}`,
      {
        env: {
          PATH: process.env.PATH,
          SNYK_TOKEN: process.env.SNYK_TOKEN_TEST,
          SNYK_API: process.env.SNYK_API_TEST,
          SNYK_LOG_PATH: logPath,
        },
      },
      (err, stdout, stderr) => {
        if (err) {
          throw err;
        }
        expect(stderr).toMatch('All requested organizations failed to be created. Review the errors in');
        expect(err).toBeNull();
        expect(stdout).toEqual('');
        done();
      },
    );
  });

  it('Fail to create orgs in strict mode when org already exists ', async (done) => {
    const pathToBadJson = path.resolve(
      __dirname + '/fixtures/create-orgs/fails-to-create/1-org.json',
    );
    const logPath = path.resolve(
      __dirname + '/fixtures/create-orgs/fails-to-create/',
    );
    return exec(
      `node ${main} orgs:create --file=${pathToBadJson} --noDuplicateNames`,
      {
        env: {
          PATH: process.env.PATH,
          SNYK_TOKEN: process.env.SNYK_TOKEN_TEST,
          SNYK_API: process.env.SNYK_API_TEST,
          SNYK_LOG_PATH: logPath,
        },
      },
      (err, stdout, stderr) => {
        if (err) {
          throw err;
        }
        expect(stderr).toMatch('All requested organizations failed to be created. Review the errors in');
        expect(err).toBeNull();
        expect(stdout).toEqual('');
        done();
      },
    );
  });
});
