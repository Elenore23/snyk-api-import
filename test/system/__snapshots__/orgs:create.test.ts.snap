// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`snyk-api-import help <...>\` Shows help text as expected 1`] = `
"index.js orgs:create

Create the organizations in Snyk based on data file generated with \`orgs:data\`
command. Output generates key data for created and existing organizations for
use to generate project import data.

Options:
  --version                      Show version number                   [boolean]
  --help                         Show help                             [boolean]
  --file                         Path to data file generated with \`orgs:data\`
                                 command                              [required]
  --noDuplicateNames             Skip creating an organization if the given name
                                 is already taken within the Group.
  --includeExistingOrgsInOutput  Log existing organization information as well
                                 as newly created                [default: true]"
`;
